### PR TITLE
Hopefully resolved hash mismatch issues and implemented suggestion

### PR DESCRIPTION
**Pull request by HPsaucii**
implemented electrons suggestion regarding the nix sections, and "fixed" the hash mismatch issues

### CODE REVIEW

**Review by isabelroses — COMMENTED** [review]
Hey no need to deal with `rec` and any from of string concationation either.

And to remove the `rec` you can declare `pname` within the `let` `in` synatx.

To remove the string concationation you can do: 
`lockFile = "${src}/Cargo.lock";` can be `lockFile = ./Cargo.lock;`

File: flake.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
   description = "A flake for MagmaWM";
 
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs";
+    nixpkgs.url = "github:nixos/nixpkgs/3af282374194ad97fd9642e7245f6264631282a2";
```
Comment (isabelroses) [inline flake.nix:5]: No need to pin nixpkgs.
```suggestion
    nixpkgs.url = "github:nixos/nixpkgs";
```

File: default.nix [hunk 2]
```diff
@@ -28,7 +25,7 @@ pkgs.rustPlatform.buildRustPackage rec {
   ];
 
   cargoLock = {
-    lockFile = "${src}/Cargo.lock";
+    lockFile = "./Cargo.lock";
```
Comment (isabelroses) [inline default.nix:28]: no need for the quotes here

File: default.nix [hunk 3]
```diff
@@ -1,9 +1,6 @@
 { lib, pkgs, ...}:
 
-pkgs.rustPlatform.buildRustPackage rec {
-  pname = "magmawm";
-  version = "main";
-
+pkgs.rustPlatform.buildRustPackage {
```
Comment (isabelroses) [inline default.nix:3]: still need to declare pname and version
```nix
 lib, pkgs, ...}: let
  pname = "magmawm";
  version = "main";
in pkgs.rustPlatform.buildRustPackage {
  inherit pname version;
```

Doing it this way means you can still use `meta.mainProgram = pname`

**Review by isabelroses — APPROVED** [review]

**Comment by HPsaucii** [comment]
thanks @isabelroses for the tips and pointers :3

**Review by nixos-goddess — APPROVED** [review]

**Pull request merged**